promql/engine.go

```diff
@@ -3721,11 +3721,12 @@ func detectHistogramStatsDecoding(expr parser.Expr) {
 			if !ok {
 				continue
 			}
-			if call.Func.Name == "histogram_count" || call.Func.Name == "histogram_sum" {
+			fn := call.Func.Name
+			if fn == "histogram_count" || fn == "histogram_sum" || fn == "histogram_avg" {
 				n.SkipHistogramBuckets = true
 				break
 			}
-			if call.Func.Name == "histogram_quantile" || call.Func.Name == "histogram_fraction" {
+			if fn == "histogram_quantile" || fn == "histogram_fraction" {
 				n.SkipHistogramBuckets = false
 				break
 			}
```

Comment:
Maybe let's write this as a `switch` statement after all?

```Go
			switch call.Func.Name {
			case "histogram_count", fn == "histogram_sum", "histogram_avg":
				n.SkipHistogramBuckets = true
			case "histogram_quantile", fn == "histogram_fraction":
				n.SkipHistogramBuckets = false
			default:
				continue
			}
			break
```

Reply:
Sure, pushed implementation using a `switch`.